### ISSUE TITLE
Read mapping vs. alignment

### ISSUE THREAD

**Issue by krdav** [nedialkova-lab/mim-tRNAseq#22]
I noticed that this method uses a read mapper that is typically used to map reads to a reference genome. However, the "genome" in this case is just a small set of tRNA sequences and it therefore seems more appropriate to me to do an actual alignment to the reference instead of a mapping.

Am I missing something, beside that an alignment might be a bit slower? 

/Kristian

**Comment by nedialkova-lab** [comment]
Hi Kristian, thanks for your interest in mim-tRNAseq. It seems that you have a conceptual question rather than an issue with the code. Our reasoning for choosing SNP-aware read alignment with GSNAP is described in quite some detail in the the paper - we believe the same reasoning speaks against using e.g. BLAST, which is what I assume you are suggesting. You are of course free to give it a try and let us know how it goes!

**Comment by krdav** [comment]
Indeed, the question is conceptual and not an issue per se. I hope you are okay that I use this channel for such contact though?

I am particularly _not_ thinking of BLAST, but true a (non-heuristics) alignment of the Smith–Waterman type. I don't expect that you have tried it and I am not asking that you should, but I would appreciate if you could share your thought about it - on a conceptual level of course?

**Comment by drewjbeh** [comment]
Hi Kristian. We haven't really tested non-heuristic approaches; the weigh up, of course, was speed and computational accessibility vs some benefits to alignment accuracy. tRNAs pose a special case of alignment challenge (again, see paper for details) that we felt needed specific solutions. The major one being clustering of similar tRNA sequences (which we did not think of first, to be clear). It seems that although tRNA sequences are highly similar to an extent, with the correct clustering this issue is minimised. And then of course deconvolution of these clusters restores resolution. 
We obviously liked the idea of clustering and deconvolution, it seems to work wonderfully, and the use of GSNAP was a low-effort approach to handling SNPs induced by misincorporations.

I see you have forked the repo and have done quite a lot of work to the code - quite honestly, these are changes I've been thinking about but haven't had good enough reasons to spend time implementing (I started coding mim-tRNAseq with very little Python/coding experience, hence the lack of nice OOP practices and classes which I see you have introduced). If you want to try and test Smith-Waterman style approaches, or would like to contribute to mim-tRNAseq, I would be happy to chat privately!
